library_generation/test/generate_library_unit_tests.sh

```diff
@@ -0,0 +1,261 @@
+}
+
+get_protobuf_version_succeed_with_valid_generator_version_test() {
+  actual_version=$(get_protobuf_version "2.24.0")
+  __assertEquals "23.2" "${actual_version}"
+  rm "gapic-generator-java-pom-parent-2.24.0.pom"
+}
+
+get_protobuf_version_failed_with_invalid_generator_version_test() {
+  actual_version=$(get_protobuf_version "1.99.0")
+  __assertEquals "" "${actual_version}"
+}
+
+search_additional_protos_common_resources_test() {
+  proto_path="${script_dir}/resources/monitoring"
+  addition_protos=$(search_additional_protos)
+  __assertEquals "google/cloud/common_resources.proto" "${addition_protos}"
+}
+
+search_additional_protos_iam_test() {
+  proto_path="${script_dir}/resources/pubsub"
+  addition_protos=$(search_additional_protos)
+  __assertEquals \
+  "google/cloud/common_resources.proto google/iam/v1/iam_policy.proto" \
+  "${addition_protos}"
+}
+
+search_additional_protos_location_test() {
+  proto_path="${script_dir}/resources/firestore"
+  addition_protos=$(search_additional_protos)
+  __assertEquals \
+  "google/cloud/common_resources.proto google/cloud/location/locations.proto" \
+  "${addition_protos}"
+}
+
+search_additional_protos_iam_location_test() {
+  proto_path="${script_dir}/resources/alloydb"
+  addition_protos=$(search_additional_protos)
+  __assertEquals \
+  "google/cloud/common_resources.proto google/iam/v1/iam_policy.proto google/cloud/location/locations.proto" \
+  "${addition_protos}"
+}
+
+get_gapic_opts_with_rest_test() {
+  proto_path="${script_dir}/resources/monitoring"
+  transport="grpc"
+  rest_numeric_enums="true"
+  gapic_opts="$(get_gapic_opts)"
+  __assertEquals \
+  "transport=grpc,rest-numeric-enums,grpc-service-config=${proto_path}/monitoring_grpc_service_config.json,gapic-config=${proto_path}/monitoring_gapic.yaml,api-service-config=${proto_path}/monitoring.yaml" \
+  "${gapic_opts}"
+}
+
+get_gapic_opts_without_rest_test() {
+  proto_path="${script_dir}/resources/monitoring"
+  transport="grpc"
+  rest_numeric_enums="false"
+  gapic_opts="$(get_gapic_opts)"
+  __assertEquals \
+  "transport=grpc,grpc-service-config=${proto_path}/monitoring_grpc_service_config.json,gapic-config=${proto_path}/monitoring_gapic.yaml,api-service-config=${proto_path}/monitoring.yaml" \
+  "$gapic_opts"
+}
+
+remove_grpc_version_test() {
+  destination_path="${script_dir}/resources/monitoring"
+  cp "${destination_path}/QueryServiceGrpc_copy.java" "${destination_path}/QueryServiceGrpc.java"
+  remove_grpc_version
+  return_code=0
+  if grep -q 'value = "by gRPC proto compiler",' "${destination_path}/QueryServiceGrpc.java"; then
+    echo "grpc version is removed."
+  else
+    echo "Error: grpc version is not removed."
+    return_code=1
+  fi
+
+  rm "${destination_path}/QueryServiceGrpc.java"
+  return "${return_code}"
+}
+
+download_generator_success_with_valid_version_test() {
+  download_generator "2.24.0"
+  __assertFileExists "gapic-generator-java-2.24.0.jar"
+  rm "gapic-generator-java-2.24.0.jar"
```

Comment:
I think we can have an "`afterEach`" function that cleans up the downloaded jars/tars, even if they don't exist. It could even be useful for local development where I personally find it a bit hard to read dir contents after I call the scripts

Reply:
This could be expanded with a `git diff` to confirm we don't have any unexpected untracked files that aren't deleted by such function

Reply:
Not all tests have clean up actions so it's difficult to write a function and apply it after every test execution.

However, I'll think about it in the reformat PR.